### PR TITLE
Use bcrypt for shared secret

### PR DESCRIPTION
**Pull request by LukasReschke**
Requires https://github.com/owncloud/core/pull/22276

### CODE REVIEW

**Comment by LukasReschke** [comment]
@karlitschek @VicDeo @DeepDiver1975 Please review.

**Comment by LukasReschke** [comment]
Relevant commit is https://github.com/owncloud/updater/commit/60ee00a0bfb0d8c902b561b06706557349de7744, the other stuff is just the symfony-polyfill for PHP 5.5 which is required for the bcrypt stuff here.

File: composer.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "psr/log": "^1.0",
         "league/plates": "^3.1",
         "paragonie/random_compat": "^1.2",
-        "symfony/polyfill-php56": "^1.1"
+        "symfony/polyfill-php56": "^1.1",
+        "symfony/polyfill-php55": "^1.1"
```
Comment (PVince81) [inline composer.json]: how is this related to bcrypt ?

Comment (LukasReschke) [inline composer.json]: That's a polyfill that provides PHP 5.5 features to older versions. Including `password_verify` :)

https://github.com/symfony/polyfill

Comment (LukasReschke) [inline composer.json]: > the password_hash and password_* related functions introduced in PHP 5.5, provided by the ircmaxell/password-compat package;

**Comment by VicDeo** [comment]
:+1:
